web_poet/pages.py

```diff
@@ -68,6 +68,25 @@ async def to_item(self) -> ItemT:
         )
 
 
+class MultiLayoutPage(ItemPage[ItemT]):
+    """Base class for :ref:`multi-layout page object classes <multi-layout>`.
+
+    Subclasses must reimplement the :meth:`layout` method.
+    """
+
+    @abc.abstractmethod
+    async def layout(self) -> ItemPage[ItemT]:
+        """Return the :ref:`page object <page-objects>` to use based on the
+        received input."""
```

Comment:
```suggestion
        received input (e.g. :class:`~.HttpResponse`)."""
```

just to clarify what type of input we're talking about.